test/error_messages.coffee

```diff
@@ -1742,3 +1742,28 @@ test "#3199: error message for throw indented comprehension", ->
       x for x in [1, 2, 3]
       ^
   '''
+
+test "#2998: Indentation mismatch in object shouldn’t produce two objects", ->
+  assertErrorFormat '''
+  colors =
+     45: 'cyan'
+     52: 'sepia'
+    100: 'olive'
+    124: 'red'
+  ''', '''
+    [stdin]:4:3: error: unexpected indentation
+      100: 'olive'
+      ^
+  '''
+
+  # Former object test that is now invalid because it looks too much like a mistake.
+  # It used to test that string-keyed objects shouldn’t suppress newlines.
+  assertErrorFormat '''
+    one =
+      '>!': 3
+    six: -> 10
```

Comment:
I guess the question is what’s more likely to happen: your “contrived” example, with a function that returns `six: -> 10`, or a slightly modified version of the original example:

```coffee
known_colors =
  1: 'cyan'
  2: 'sepia'
100: 'olive'
124: 'red'
```

In other words, it could’ve been happenstance that the original poster’s “left-indented” keys didn’t stretch back to the same indentation level as the variable name, as my example here does. If we adopt the rule you suggest, then this example presents the same issue as the original example did.

Whereas if we get more strict about indentation in object literals, your contrived example would simply throw a compiler error, which would be easily resolved by putting `return` before `six:` or by wrapping the returned object in braces.